File: endpoints/photoscategories.py
from silence.decorators import endpoint



@endpoint(
    route="/photoscategories",
    method="GET",
    sql="SELECT * FROM photoscategories"
)
def get_all():
    pass
############################

@endpoint(
    route="/photoscategories/$photoCategoryId",
    method="GET",
    sql="SELECT * FROM PhotosCategories WHERE photoCategoryId=$photoCategoryId"
)
def get_by_id():
    pass

################################
@endpoint(
    route="/photoscategories/numcategoryId/$photoId",
    method="GET",
    sql="SELECT COUNT(*) as num FROM PhotosCategories WHERE photoId=$photoId",
    description="Obtain num of categoryId with photoId"
)
def get_num_by_photoId():
    pass
#################################
@endpoint(
    route="/photoscategories/byPhotoId/$photoId",
    method="GET",
    sql="SELECT * FROM PhotosCategories WHERE photoId=$photoId",
    description="Obtain num of categoryId with photoId"
)
def get_by_photoId():
    pass
####################################

@endpoint(
    route="/photoscategories/getByCategoryId/$categoryId",
    method="GET",
    sql="SELECT * FROM PhotosCategories WHERE categoryId=$categoryId",
    description="Obtain num of categoryId with photoId"
)
def get_by_categoryId():
    pass
####################################

@endpoint(
    route="/photoscategories",
    method="POST",
    sql="INSERT INTO PhotosCategories (photoId, categoryId) VALUES ($photoId, $categoryId)",
    description="Creates a new photocategory",
    auth_required=True,
)
def create(photoId, categoryId):
    pass

###############################################################################

@endpoint(
    route="/photoscategories/$photoCategoryId",
    method="DELETE",
    sql="DELETE FROM Photoscategories WHERE photoCategoryId = $photoCategoryId",
    description="Removes a photocategory",
    auth_required=True,
)
def delete():
    pass

##################################################################################

@endpoint(
    route="/photoscategories/$photoId/$categoryId",
    method="DELETE",
    sql="DELETE FROM Photoscategories WHERE photoId=$photoId AND categoryId=$categoryId",
    description="Removes a photocategory",
    auth_required=True,
)
def delete_by_photoId_categoryId():
    pass

File: docs/js/api/photoscategories.js
"use_strict"
 import { BASE_URL, requestOptions } from "./common.js";

 const photoscategoriesAPI = {
    getAll: function () {
        return new Promise(function(resolve, reject) {
            axios
                .get(`${BASE_URL}/photoscategories`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getByPhotoId: function (photoId) { //Obtiene a partir de la photoId
        return new Promise(function(resolve, reject) {
            axios
                .get(`${BASE_URL}/photoscategories/byPhotoId/${photoId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getNumByPhotoId: function (photoId) { //Get num by photoId
        return new Promise(function(resolve, reject) {
            axios
                .get(`${BASE_URL}/photoscategories/numcategoryId/${photoId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getByCategoryId: function (categoryId) { //Obtiene a partir de la id de la categoria
        return new Promise(function(resolve, reject) {
            axios
                .get(`${BASE_URL}/photoscategories/getByCategoryId/${categoryId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    create: function(formData) {
        return new Promise (function(resolve,reject) {
            axios
                .post(`${BASE_URL}/photoscategories`,formData ,requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    deleteByPhotoIdCategoryId: function(photoId, categoryId) {
        return new Promise (function(resolve,reject) {
            axios
                .delete(`${BASE_URL}/photoscategories/${photoId}/${categoryId}`, requestOptions
                )
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    }

    
 };

 export { photoscategoriesAPI };